src/eip7594/eip7594.c

```diff
@@ -524,12 +523,23 @@ static C_KZG_RET compute_commitment_to_aggregated_interpolation_poly(
             blst_fr_add(
                 &aggregated_column_cells[index], &aggregated_column_cells[index], &scaled_fr
             );
-
-            /* Mark the cell as being used */
-            is_cell_used[index] = true;
         }
     }
 
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    // Determine which cells are used
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+
+    /* Start with false values */
+    for (size_t i = 0; i < CELLS_PER_EXT_BLOB; i++) {
+        is_cell_used[i] = false;
+    }
+
+    /* Mark each cell index as used */
+    for (uint64_t i = 0; i < num_cells; i++) {
```

Comment:
Why is `i` a `size_t` above, and a `uint64_t` here? Any particular reason?

Reply:
Yeah it's a bit of an odd situation. It's because of `num_cells` type. `num_cells` is a `uint64_t` because it comes from the public API while we try to use `size_t` internally.

Reply:
ok!